src/Microsoft.CodeQuality.Analyzers/Core/QualityGuidelines/AvoidPropertySelfAssignment.cs

```diff
@@ -0,0 +1,72 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Immutable;
+using Analyzer.Utilities;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Operations;
+
+namespace Microsoft.CodeQuality.Analyzers.QualityGuidelines
+{
+    /// <summary>
+    /// CA1099: Prevent properties from being assigned to themselves
+    /// </summary>
+    [DiagnosticAnalyzer(LanguageNames.CSharp, LanguageNames.VisualBasic)]
+
+    public sealed class AvoidPropertySelfAssignment : DiagnosticAnalyzer
+    {
+        internal const string RuleId = "CA1099";
```

Comment:
TODO (for myself): Ensure we assign the correct category and next available diagnostic ID in that category. I will get back to you on this soon.

Reply:
@mclayton7 I reviewed different rule sections under https://docs.microsoft.com/en-us/visualstudio/code-quality/code-analysis-for-managed-code-warnings. This rule seems to be more suitable to be under `DiagnosticCategory.Usage`, and next available RuleId in that category is `CA2245`. Can you please update your analyzer accordingly?